src/firefly/js/templates/router/RouteHelper.jsx

```diff
@@ -111,10 +111,11 @@ export function useDropdownRoute() {
     useEffect(() => {
         const {path, visible} = getViewInfo();
         if (pathname !== path) {
+            const {menuItems,showBgMonitor}= getMenu();
             const menuItem = getMenuItem(pathname);
             if (menuItem) {     // the requested pathname is in the menu
                 dispatchSetLayoutInfo({dropDown:{view: menuItem.action, menuItem, visible: true}});
-                dispatchSetMenu({selected: menuItem.action});
+                dispatchSetMenu({selected: menuItem.action, menuItems, showBgMonitor});
```

Comment:
@loitly for when you're back on Monday, you may want to take a look at this fix. Since we would call `dispatchSetMenu` with only the selected tab when you refresh the page, the app would display that tab, but `menuItems` previously existing would get overwritten (you can try that behavior in Euclid or SPherex on nightly). So I'm passing them explicitly here again. 

The reason job monitor tab still showed up is because the `MENU_UPDATE` action creator would call this after `dispatchSetMenu`, which checks for adding background monitor as a menu item.

Reply:
I think you just want to select the menu. It was working before because `dispatchSetMenu` allowed partial updates, but that caused issues elsewhere. We've recently fixed it so that it now performs a full 'set'.
If you want to select a menu item, you should use `LayoutCntlr.dispatchShowDropDown` instead.  Can you try this to see if that fixes the problem?